apps/examples/testcase/le_tc/network/itc_net_connect.c

```diff
@@ -0,0 +1,332 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file itc_net_connect.c
+/// @brief ITC Scenario Test Case for connect() API
+
+#include <sys/socket.h>
+#include <string.h>
+#include <unistd.h>
+#include <pthread.h>
+#include "tc_internal.h"
+
+#define SERVER_MSG "Hello from server"
+#define CLIENT_MSG "Hello from client 0 "
+#define BUFF_LEN 64
+#define SERVER_WAIT_TIME 3
+#define CLIENT_WAIT_TIME 7
+#define OK 0
+#define ERROR -1
+
+static int g_port;
+/**
+* @fn               :server_connect
+* @description      :starts server
+* @return           :void
+*/
+static void *server_connect(void *ptr_num_clients)
+{
+	int server_socket;
+	int client_socket;
+	int valread = 0;
+	int valsend = 0;
+	int opt = 1;
+	int ret;
+	struct sockaddr_in address;
+	int addrlen = sizeof(address);
+	int num_clients = *((int *)ptr_num_clients);
+	char buffer[BUFF_LEN] = {0};
+	char *ptr_msg = SERVER_MSG;
+	int total_read;
+	int total_send;
+	int pret = OK;
+	int read_len = 0;
+	int send_len = 0;
+
+	printf("Server thread started...\n");
+	server_socket = socket(AF_INET, SOCK_STREAM, 0);
+	if (server_socket < 0) {
+		pret = ERROR;
+		return (void *)pret;
+	}
+	setsockopt(server_socket, SOL_SOCKET, SO_REUSEADDR | SO_REUSEPORT, &opt, sizeof(opt));
+	address.sin_family = AF_INET;
+	address.sin_addr.s_addr = INADDR_ANY;
+	address.sin_port = htons(g_port);
+	ret = bind(server_socket, (struct sockaddr *)&address, sizeof(address));
+	if (ret != OK) {
+		close(server_socket);
+		pret = ERROR;
+		return (void *)pret;
+	}
+	ret = listen(server_socket, 5);
+	if (ret != OK) {
+		close(server_socket);
+		pret = ERROR;
+		return (void *)pret;
+	}
+	while (num_clients-- > 0) {
+		client_socket = accept(server_socket, (struct sockaddr *)&address, (socklen_t*)&addrlen);
+		if (client_socket < 0) {
+			close(server_socket);
+			pret = ERROR;
+			return (void *)pret;
+		}
+		total_read = strlen(CLIENT_MSG);
+		memset(buffer, 0, BUFF_LEN);
+		while (total_read) {
+			valread = read(client_socket, buffer + read_len, BUFF_LEN - read_len);
```

Comment:
I think that using recv() is better than read()

Reply:
changes  are done as per your suggestion.  please review